### PR TITLE
Refactor onnx graph ops function & registry

### LINKED ISSUE #95: [FEATURE] Refactor onnx graph ops function and registry
**Is your feature request related to a problem? Please describe.**
Existing graph ops represented as function only and the graph registry method is clunky. This makes graph ops less flexible when calling/creating graph_ops outside exporter. For example we want to specify list of ops to be performed on model, from another function/class.

**Describe the solution you'd like**
Represent graph ops as callable class, clean up graph ops registry.

### PR DESCRIPTION
**Pull request by alifahrri**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- close #95 

Related issues : 

## What is the new behavior?

-
-
-

## Checklist

- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/vortex/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by triwahyuu** [comment]
there is a good implementation for registry in [mmdetection](https://github.com/open-mmlab/mmdetection) which use [mmcv](https://github.com/open-mmlab/mmcv) here https://github.com/open-mmlab/mmcv/blob/master/mmcv/utils/registry.py

example usage: 
- https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/builder.py
- https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/detectors/yolo.py

might be a nice reference implementation for a more general registry

**Comment by alifahrri** [comment]
> there is a good implementation for registry in [mmdetection](https://github.com/open-mmlab/mmdetection) which use [mmcv](https://github.com/open-mmlab/mmcv) here https://github.com/open-mmlab/mmcv/blob/master/mmcv/utils/registry.py
> 
> example usage:
> 
>     * https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/builder.py
> 
>     * https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/detectors/yolo.py
> 
> 
> might be a nice reference implementation for a more general registry

ok

**Comment by triwahyuu** [comment]
maybe in different PR for that, I have temporarily implement that for model registry in pytorch lightning integration

**Comment by alifahrri** [comment]
> maybe in different PR for that, I have temporarily implement that for model registry in pytorch lightning integration

ok

File: src/development/vortex/development/exporter/utils/onnx/graph_ops/base_ops.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import onnx
+
+class GraphOpsBase:
+    def __init__(self):
+        """Base Class to represents onnx graph ops
+        """
+        pass
+
+    def run(self, model: onnx.ModelProto) -> onnx.ModelProto:
```
Comment (triwahyuu) [inline src/development/vortex/development/exporter/utils/onnx/graph_ops/base_ops.py:9]: what do you think about using [`abstractmethod`](https://docs.python.org/3/library/abc.html#abc.abstractmethod)? I just realized we can actually enforce abstract function in python

Comment (alifahrri) [reply src/development/vortex/development/exporter/utils/onnx/graph_ops/base_ops.py:9]: something like this?
```
class Base(metaclass=ABCMeta):
   ...
   @abc.abstractmethod
   def run(...)
```

Comment (alifahrri) [reply src/development/vortex/development/exporter/utils/onnx/graph_ops/base_ops.py:9]: ok please wait

Comment (triwahyuu) [reply src/development/vortex/development/exporter/utils/onnx/graph_ops/base_ops.py:9]: yepp

**Comment by triwahyuu** [comment]
LGTM, sorry I accidentally closed the pr

**Pull request merged**